src-ts/proposalCreator.ts

```diff
@@ -125,11 +125,29 @@ export class RoundTripProposalCreator {
     public readonly l1Config: L1GovConfig,
     public readonly targetNetworkConfig: UpgradeConfig
   ) {}
-  
-  /**
+
+    /**
    * Creates calldata for roundtrio path; data to be used either in a proposal or directly in timelock.schedule
    */
-  private async createRoundTripCallData(   
+  public async createRoundTripCallData(   
+    upgradeAddr: string,
+    upgradeValue: BigNumber,
+    upgradeData: string,
+    proposalDescription: string){
+
+      const {l1TimelockTo, l1TimelockScheduleCallData } = await  this.createArbSysArgs(upgradeAddr, upgradeValue, upgradeData, proposalDescription) 
+
+      const iArbSys = ArbSys__factory.createInterface();
+      return iArbSys.encodeFunctionData("sendTxToL1", [
+        l1TimelockTo,
+        l1TimelockScheduleCallData,
+      ]);
+      
+    }
+  /**
+   * Generates arguments for ArbSys.sendTxToL1 for a constitutional proposal. Can be used to submit a proposal in e.g. the Tally UI.  
+  */
+  public async createArbSysArgs(   
```

Comment:
I think this is a bit of a confusing name, I think we need to capture more that this creates all the round trip proposal up to the arb sys call.